appng-core/src/main/java/org/appng/core/model/RepositoryCacheBase.java

```diff
@@ -77,8 +80,22 @@ public void reload() throws BusinessException {
 	abstract void update(String packageName) throws BusinessException;
 
 	public List<PackageWrapper> getApplications() throws BusinessException {
+		return getApplications(null);
+	}
+
+	public List<PackageWrapper> getApplications(String packageName) throws BusinessException {
 		update();
-		List<PackageWrapper> applications = new ArrayList<PackageWrapper>(applicationWrapperMap.values());
+		List<PackageWrapper> applications = new ArrayList<>(applicationWrapperMap.values());
+		if (StringUtils.isNotBlank(packageName)) {
+			Predicate<? super PackageWrapper> filter;
+			String wildcard = "*";
+			if (packageName.contains(wildcard)) {
+				filter = p -> p.getName().matches(packageName.replace(wildcard, ".*?"));
+			} else {
+				filter = p -> p.getName().startsWith(packageName);
+			}
+			applications = applications.parallelStream().filter(filter).collect(Collectors.toList());
```

Comment:
Whats about the order? The order can be different by using `parallelStream()` this maybe causes test errors because of unstable results

Reply:
Collection is sorted afterwards anyways.

Reply:
Right, I didn't saw that.